### PR TITLE
fossa upload-project command

### PR DESCRIPTION
**Pull request by zlav**
## Description

Add support for a `fossa upload-project` command that is intended to behave identically to the UI archive scanning feature.

This PR also adds support to assign a project to a release group version from the CLI by supplying `--release-group <name>` and `--release-group-version <version>` to the command.

Example UI:
```
fossa archive temp --title new-title --release-group test --release-group-version 2.0
============================================================

    FOSSA Reports:
    https://app.fossa.com/projects/archive+testing%2F/refs/%2Ffossa-archive/6b15f78939019bba6ead53cd2d3a4b8d

============================================================
```

## Notes
I considered using one flag and splitting on a special character such as `--release-group <name>@<version>` but I decided to implement two separate flags. The main reason is to avoid unintended issues with splitting the strings on the backend and disallowing a certain special character. The second reason is to reduce confusion for the user who needs to manually add the character. I am favoring enforcing explicit declaration over the expense of the command looking a bit messy.

### CODE REVIEW

File: api/fossa/locator.go
Comment (liftM) [inline api/fossa/locator.go:72]: Huh. How did we not do this before? Was this just broken before?

File: api/fossa/tar.go
Comment (liftM) [inline api/fossa/tar.go:67]: Non-blocking given urgency, but it would be better style here for `UploadTarball` to take an options struct e.g.

```go
type UploadTarballOptions struct {
  Name string
  Revision string
  // ...
}
```

This helps people avoid accidentally mix up the order of arguments with the same type (e.g. `name` and `revision`, or `rawLicenseScan` and `upload`). Besides, you're taking an `UploadOptions` struct anyway.

File: api/fossa/tar.go
Comment (liftM) [inline api/fossa/tar.go:443]: Why this remove the name here? I would prefer that struct values in slices are named, to reduce ambiguity. Is there an argument for not naming them?

File: config/file.v1/file.go
Comment (liftM) [inline config/file.v1/file.go:37]: [nit] Hmm, I remember deciding that we preferred the compact syntax `--release_group name@version`. I don't think I have a strong opinion either way. If this is done and works and is simpler, let's go for it.

Just for posterity's sake, was there a particular reason you decided against the compact syntax?

File: cmd/fossa/cmd/upload_project/upload_project.go
Comment (liftM) [inline cmd/fossa/cmd/upload_project/upload_project.go:18]: I would be more explicit: "Uploads a module's source code as a top-level FOSSA project rather than a dependency."

**Review by liftM — APPROVED** [review]
LGTM. Some comments. Non-blocking due to urgency - please respond before merging, but don't feel like you need to block on a re-approval before merge.

File: api/fossa/locator.go
Comment (zlav) [reply api/fossa/locator.go:72]: No, we never did this previously but I don't think it was really "broken" because we've never supported top-level projects with `archive` locators through the CLI until now.

File: api/fossa/tar.go
Comment (zlav) [reply api/fossa/tar.go:443]: I changed this because I got a linting warning and didn't have a preference either way, but if you see a benefit in keeping it I'm happy to do that. I believe the linter warns because the explicit `Component` is redundant as we are already defining the slice as `[]Component`.

File: config/file.v1/file.go
Comment (zlav) [reply config/file.v1/file.go:37]: Yea good question, this is a note I left in the PR
```
I considered using one flag and splitting on a special character such as --release-group <name>@<version> but I decided to implement two separate flags. The main reason is to avoid unintended issues with splitting the strings on the backend and disallowing a certain special character. The second reason is to reduce confusion for the user who needs to manually add the character. I am favoring enforcing explicit declaration over the expense of the command looking a bit messy.
```
I understand some of the issues with having two flags, but I think being explicit is better in the long run as long as we fail when both aren't supplied.

File: api/fossa/tar.go
Comment (zlav) [reply api/fossa/tar.go]: Yea I like this, I'll implement this here.

**Pull request merged**
